### PR TITLE
cleanup and docs update, string types descriptive

### PR DESCRIPTION
**Pull request by tabaktoni**
Fix https://github.com/0xs34n/starknet.js/issues/742
Fix https://github.com/0xs34n/starknet.js/issues/724

## Motivation and Resolution
- [x] Secure cairoVersion from non-ts unintentional number usage
- [x] Removed unused constants that are now part of ec.
```ts
export const FIELD_PRIME = '800000000000011000000000000000000000000000000000000000000000001';
export const FIELD_GEN = '3';
export const FIELD_SIZE = 251;
export const EC_ORDER = '800000000000010FFFFFFFFFFFFFFFFB781126DCAE7B2321E66A241ADC64D2F';
export const ALPHA = '1';
export const BETA = '6F21413EFBE40DE150E596D72F7A8C5609AD26C15C915C1F4CDFCB99CEE9E89';
export const MAX_ECDSA_VAL = '800000000000000000000000000000000000000000000000000000000000000';
export const CONSTANT_POINTS = [...]
```
- [x] Document utils methods
- [x] add string type to documentation
- [x] cleanup

## Checklist:

- [ ] Performed a self-review of the code
- [ ] Rebased to the last commit of the target branch (or merged it into my branch)
- [ ] Linked the issues which this PR resolves
- [ ] Documented the changes in code (API docs will be generated automatically)
- [ ] Updated the tests
- [ ] All tests are passing

### CODE REVIEW

**Review by penovicp — COMMENTED** [review]
Noticed that some of the JSDOC/TSDOC comment elements clash with the autogenerated content. I'll open a PR targeting this branch with the corresponding tweaks.

File: src/utils/calldata/cairo.ts [hunk 1]
```diff
@@ -61,13 +60,19 @@ export function isCairo1Abi(abi: Abi): boolean {
 }
 
 /**
- * named tuple are described as js object {}
- * struct types are described as js object {}
- * array types are described as js array []
+ * named tuple cairo type are described as js object {}
```
Comment (ivpavici) [inline src/utils/calldata/cairo.ts:63]:
```suggestion
 * named tuple cairo type is described as js object {}
```
and for below

File: src/utils/shortString.ts [hunk 2]
```diff
@@ -1,50 +1,77 @@
...
- * @param str - ASCII string -
- * 31 characters maxi. Ex : "uri/item23.jpg"
+ * @param str string - short string (ASCII string 31 characters max)
  * @returns a string representing an Hex number 248 bits max.
  * @Example
  * ```typescript
  * const myEncodedString: string = encodeShortString("uri/pict/t38.jpg");
  * ```
- * returns : string : "0x7572692f706963742f7433382e6a7067"
+ * return hex string (ex."0x7572692f706963742f7433382e6a7067")
```
Comment (ivpavici) [inline src/utils/shortString.ts:74]:
```suggestion
 * return hex string (ex. "0x7572692f706963742f7433382e6a7067")
```

File: src/utils/shortString.ts [hunk 3]
```diff
@@ -54,15 +81,13 @@ export function encodeShortString(str: string): string {
...
- * Ex : hex : "0x1A4F64EA56" or decimal : "236942575435676423"
- * @returns a string with 31 characters max.
+ * @param str string - representing a 248 bits max number. (ex."0x1A4F64EA56" or "236942575435676423")
+ * @returns string - short string (a string with 31 characters max.)
  * @Example
  * ```typescript
  * const myDecodedString: string = decodeShortString("0x7572692f706963742f7433382e6a7067");
  * ```
- * return : string : "uri/pict/t38.jpg"
+ * return string (ex."uri/pict/t38.jpg")
```
Comment (ivpavici) [inline src/utils/shortString.ts:90]:
```suggestion
 * return string (ex. "uri/pict/t38.jpg")
```

**Review by penovicp — APPROVED** [review]

**Review by ivpavici — APPROVED** [review]

**Pull request merged**